storage/innobase/row/row0merge.cc

```diff
@@ -3561,10 +3568,15 @@ in field data for the tuple
 @param tuple     tuple to be inserted
 @param heap      heap to allocate the memory for the blob storage
 @param blob_file file to handle blob data */
-static dberr_t row_merge_copy_blob_from_file(dtuple_t *tuple, mem_heap_t *heap,
+static dberr_t row_merge_copy_blob_from_file(dtuple_t *tuple,
+                                             dict_index_t *index,
+                                             mem_heap_t *heap,
                                              merge_file_t *blob_file)
 {
-  for (ulint i = 0; i < dtuple_get_n_fields(tuple); i++)
+  const uint blob_prefix= dict_table_has_atomic_blobs(index->table)
+                          ? 0
+                          : REC_ANTELOPE_MAX_INDEX_COL_LEN;
```

Comment:
The `?` and `:` lines should be indented only by 2 extra spaces.

Can we assert `ut_ad(tuple->n_fields == index->n_fields)`?